### PR TITLE
Remove unnecessary pragmas

### PR DESCRIPTION
**Pull request by amcasey**
...now that https://github.com/dotnet/linker/issues/2715 has been fixed.

### CODE REVIEW

File: src/Shared/PropertyHelper/PropertyHelper.cs [hunk 1]
```diff
@@ -212,8 +212,6 @@ public void SetValue(object instance, object? value)
         // MakeGenericMethod + value type requires IsDynamicCodeSupported to be true.
         if (RuntimeFeature.IsDynamicCodeSupported)
         {
-            // TODO: Remove disable when https://github.com/dotnet/linker/issues/2715 is complete.
-#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
```
Comment (eerhardt) [inline src/Shared/PropertyHelper/PropertyHelper.cs:216]: There are a few more places, can we remove them as well?

https://github.com/dotnet/aspnetcore/blob/12e4c649c37e61ea34be2b4b86dc5c7473860fe5/src/Http/Routing/src/Matching/JumpTableBuilder.cs#L92-L93

https://github.com/dotnet/aspnetcore/blob/12e4c649c37e61ea34be2b4b86dc5c7473860fe5/src/Http/Http.Results/src/ResultsOfTHelper.cs#L50-L51

Comment (amcasey) [reply src/Shared/PropertyHelper/PropertyHelper.cs:216]: Hooray for vscode.dev!

**Review by eerhardt — COMMENTED** [review]
Fix build

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**